[WIP] Add GitLab Runner package.

ssk2 opened the pull request.

ssk2 committed:
Add GitLab CI package.

ssk2 committed:
Remove selected.

ssk2 committed:
Update name to be executor.

ssk2 committed:
Update required.

ssk2 committed:
Escape environment variables values.

ssk2 committed:
Add privileged flag.

ryadav88 commented:
@tobilg can you update this PR to add your changes for Gitlab Runner?

tobilg commented:
I created another PR for merging my changes into the `gitlab-ci` branch, see https://github.com/mesosphere/universe/pull/843

tobilg committed:
Addition of Gitlab Runner (#843)

* Multiple changes:

- Renamed service to `gitlab-runner`
- Updated GitLab Runner version to 1.7.1
- Some additions to the `package.json`

* Added missing newline to `config.json`

* Deletion of `gitlab-ci` folder in favor of `gitlab-runner`

ryadav88 commented:
@tobilg ready to merge once we have an `example` in https://github.com/dcos/examples

ryadav88 commented on repo/packages/G/gitlab-runner/0/resource.json:
```diff
@@ -0,0 +1,14 @@
+  "images": {
+    "icon-small": "https://secure.gravatar.com/avatar/6edd0acaf80f784fab3dd2c31d604e74.jpg?s=40&r=g&d=mm",
+    "icon-medium": "https://secure.gravatar.com/avatar/6edd0acaf80f784fab3dd2c31d604e74.jpg?s=80&r=g&d=mm",
+    "icon-large": "https://secure.gravatar.com/avatar/6edd0acaf80f784fab3dd2c31d604e74.jpg?s=200&r=g&d=mm"
```
use the `gitlab runner` logo instead: https://gitlab.com/uploads/project/avatar/250833/runner_logo.png

ryadav88 commented on repo/packages/G/gitlab-runner/0/package.json:
```diff
@@ -0,0 +1,22 @@
+  "maintainer": "support@gitlab.com",
+  "website": "https://about.gitlab.com/gitlab-ci/",
+  "framework": false,
+  "description": "GitLab Runner is the open-source continuous integration (CI) service included with GitLab that coordinates the testing.",
```
add an example in dcos/examples and link it at the end of `description`

tobilg replied:
Will do.

tobilg committed:
Gitlab Runner package: Update version and add Docker-related properties (#864)

* Added the possibility to use insecure registry, and define default Docker image

* Bump GitLab Runner version to 1.8.1

tobilg committed:
Gitlab CI bugfixes & updates (#867)

* Added the possibility to use insecure registry, and define default Docker image

* Bump GitLab Runner version to 1.8.1

* Fix Mustache logic for Docker-related properties

* Set correct version

* Add further descriptions

ssk2 committed:
Append executor name to Marathon service name.

ssk2 committed:
Update Docker image to Mesosphere repo.

joerg84 commented:
@ssk2 is this still WIP?

ssk2 commented:
@joerg84: yes, that is why we haven't merged it. Awaiting feedback from GitLab's engineers.

markpundsack commented:
@ssk2 Sorry, I'm just jumping in here. After the feature freeze for our next release, a GitLab engineer will review and provide feedback.

tobilg commented:
Hi @markpundsack, thanks for the heads-up! When will be the feature freeze?

markpundsack commented:
@tobilg Feature freeze is today. :)

ssk2 committed:
Add healthcheck.

ssk2 committed:
Update constraint property.

ssk2 committed:
Update description and default.

ssk2 committed:
Reduce minimum memory.

tmaczukin commented:
@ssk2 @tobilg I've got back to test this, but since todays morning I have problems with setting up DC/OS cluster. When I'll resolve this problem I'll test the integration again and leave a feedback for you.

ryadav88 commented:
@tobilg @ssk2 any update here?

tobilg commented:
Waiting for the https://github.com/mesosphere/universe/pull/1126 to be merged, then testing with the Runner version 9.1

ayufan commented:
OK. So we are currently waiting for https://github.com/mesosphere/universe/pull/1126?
